### PR TITLE
Compatibility with PrestaShop 8.x

### PR DESCRIPTION
**Pull request by GytisZum**
Changed all deprecated method into new one by implementing version utility class. This class checking current Ps version and comparing.

- Added upgrade method which will change deprecated hook.
- Added `.gitignore` file to ignore vendor file inside repository
- Absence or deprecation of certain methods.  `JsonEncode` method does not exist in PS8
- In back office no error messages is shown.
- Some of the responses do not have valid values which leads to error and improper functionality fix.
- Module trying to access non-existent values fix.

### CODE REVIEW

File: controllers/admin/AdminSmartsuppAjax.php [hunk 1]
```diff
@@ -32,26 +32,36 @@ public function init()
...
                     'password' => Tools::getValue('password'),
                     'platform' => 'Prestashop ' . _PS_VERSION_,
-                ));
+                ]);
 
-                Configuration::updateValue('SMARTSUPP_KEY', $response['account']['key']);
-                Configuration::updateValue('SMARTSUPP_EMAIL', Tools::getValue('email'));
+                if (!isset($response['account']['key'])) {
+                    $response['error'] = isset($response['error']) ? $response['error'] : $this->module->l('Unknown Error Occurred');
+                    $response['message'] = isset($response['message']) ? $response['message'] : $this->module->l('Unknown Error Occurred');
```
Comment (tomjas1997) [inline controllers/admin/AdminSmartsuppAjax.php]: self::FILE_NAME for translatable text

Comment (tomjas1997) [reply controllers/admin/AdminSmartsuppAjax.php]: to skip usage of else, might as well break after filling in the response. Would make this a early return

File: controllers/admin/AdminSmartsuppAjax.php [hunk 2]
```diff
@@ -65,12 +75,24 @@ public function init()
...
 
         header('Content-Type: application/json');
-        die(Tools::jsonEncode(array(
-                    'key' => Configuration::get('SMARTSUPP_KEY'),
-                    'email' => Configuration::get('SMARTSUPP_EMAIL'),
-                    'error' => $response['error'],
-                    'message' => $response['message'],
-                    'hint' => $response['hint']
-                )));
+
```
Comment (tomjas1997) [inline controllers/admin/AdminSmartsuppAjax.php:61]:
```suggestion
$responseData = [
            'key' => Configuration::get('SMARTSUPP_KEY'),
            'email' => Configuration::get('SMARTSUPP_EMAIL'),
            'error' => isset($response['error']) ? $response['error'] : null.
        ];
        
        $responseData = array_filter($responseData, function ($val) {
            return $val !== null;
        })
```

File: controllers/admin/AdminSmartsuppAjax.php [hunk 3]
```diff
@@ -32,26 +32,36 @@ public function init()
...
                     'consentTerms' => 1,
                     'platform' => 'Prestashop ' . _PS_VERSION_,
-                ));
+                ]);
 
-                Configuration::updateValue('SMARTSUPP_KEY', $response['account']['key']);
-                Configuration::updateValue('SMARTSUPP_EMAIL', Tools::getValue('email'));
+                if (!isset($response['account']['key'])) {
+                    $response['error'] = isset($response['error']) ? $response['error'] : $this->module->l('Unknown Error Occurred');
+                    $response['message'] = isset($response['message']) ? $response['message'] : $this->module->l('Unknown Error Occurred');
```
Comment (zuk3975) [inline controllers/admin/AdminSmartsuppAjax.php]: $this->module->l(`Unknown Error Occurred`) could be reused (already used four times)

File: smartsupp.php [hunk 4]
```diff
@@ -312,20 +334,27 @@ public function hookHeader()
...
-                $js .= '<script type="text/javascript" src="'.$path.'views/js/smartsupp.js"></script>';
-                $js .= '<link rel="stylesheet" href="'.$path.'views/css/smartsupp.css" type="text/css" />';
-                $js .= '<link rel="stylesheet" href="'.$path.'views/css/smartsupp-nobootstrap.css" type="text/css" />';
-            }
+            $js .= '<script type="text/javascript" src="' . $path . 'views/js/smartsupp.js"></script>';
+            $js .= '<link rel="stylesheet" href="' . $path . 'views/css/smartsupp.css" type="text/css" />';
+            $js .= '<link rel="stylesheet" href="' . $path . 'views/css/smartsupp-nobootstrap.css" type="text/css" />';
+        }
+
+        return $js;
```
Comment (tomas862) [inline smartsupp.php:345]: would be much better to return tpl file with this code

Comment (GytisZum) [reply smartsupp.php:345]: This work is not in current scope, there is many html code inside project and should be done seperatly

**Pull request merged**